modules/lnd/lnd_wrapper/wrapper.go

```diff
@@ -130,4 +137,30 @@ func LndDeserializeInvoice(cInvoiceStr *C.char) *C.char {
 	return C.CString(sb.String())
 }
 
+//export LndParseP2pLightningMessage
+func LndParseP2pLightningMessage(data C.ByteArray) *C.char {
+	buffer := C.GoBytes(unsafe.Pointer(data.data), data.length)
```

Comment:
We package the pointer and size into a `ByteArray`, then we extract the pointer and size here anyway.  It would be simpler to get rid of `ByteArray` entirely and pass the pointer and size directly to this function.